overlay.nix

```diff
@@ -31,7 +31,11 @@ in final: prev: {
           then "glibc-2.30"
           else final.glibc.name;
         inherit sources;
-        ghcVersion = args.ghc.version;
+        # compiler-nix-name is of the form 'ghc883'
+        ghcVersion = if args ? compiler-nix-name then
+          builtins.concatStringsSep "."
+            (builtins.match "ghc([0-9])([0-9])([0-9])" args.compiler-nix-name)
+                     else args.ghc.version; # deprecated
```

Comment:
Unfortunately this won't work because `compiler-nix-names` looks like `ghc8101` for 8.10.1, which is (almost) ambiguous. Instead you can get the version without parsing `compiler-nix-name` from `final.haskell-nix.compiler.${compiler-nix-name}.version`

Reply:
You know what they say about regexes...